### ISSUE TITLE
Add new ante-dependence covariance structure

### ISSUE THREAD

**Issue by danielinteractive** [glmmTMB/glmmTMB#810]
Hi all,

we have situations where an ante-dependence covariance structure would be useful. 
(See e.g. https://documentation.sas.com/doc/en/pgmsascdc/9.4_3.3/statug/statug_mixed_syntax14.htm which 
defines:
![image](https://user-images.githubusercontent.com/1071227/167382488-cf93a4b1-d651-4be9-90e9-831a77b21893.png)

I guess from looking at https://github.com/glmmTMB/glmmTMB/blob/286958f5c797c0154a912f2a57b3c38e96d13d1a/glmmTMB/src/glmmTMB.cpp#L238
that we would need to add corresponding C++ block there, how is it on the R side - do you maybe have a developer guidance / previous PR where we can see which boxes to tick when adding a new covariance structure to `glmmTMB`?

Thanks,
cheers
Daniel

**Comment by kaskr** [comment]
There's some information in this old PR:

https://github.com/glmmTMB/glmmTMB/pull/195

It appears that the Toeplitz structure was added by these commits:

https://github.com/glmmTMB/glmmTMB/pull/195/commits/d91fb5baad99bb73ed8e33a4316a752e4f808734
https://github.com/glmmTMB/glmmTMB/pull/195/commits/b35df487938273dcdc11a1f38965516ab67c95c6

and that one has to not forget to run `make enum-update`.

**Comment by kaskr** [comment]
BTW, note that one more thing to consider before implementing a covariance structure is computational complexity. Based on its similarity to AR1, I'm guessing the ante dependence structure could be given a state space formulation (linear complexity) rather than covariance formulation (cubic complexity). If the model is expected to be used with long time series data (say more than dimension 100) this may be worth considering - otherwise probably not.

**Comment by danielinteractive** [comment]
thanks @kaskr for the pointers and tipps! cool.
yeah let me see if there is a state space formulation, not sure. Luckily we don't have more than 100 time points in our case for sure, but it would be nice if this would be usable even in such cases I guess.

**Comment by danielinteractive** [comment]
Just found this page:

![image](https://user-images.githubusercontent.com/1071227/167413977-9d23af8f-a324-42d4-b6b1-88cf1fc0ef4f.png)

in the book "Longitudinal Data with Serial Correlation: A State-Space Approach" so I guess that's it

**Comment by kaskr** [comment]
Yeah, that's what I expected. The `phi`s are time varying auto regressive coefficients.
I think this could be implemented very similar to how it was done for [AR1](https://github.com/glmmTMB/glmmTMB/blob/286958f5c797c0154a912f2a57b3c38e96d13d1a/glmmTMB/src/glmmTMB.cpp#L261).
